lib/strings.nix

```diff
@@ -1,183 +1,263 @@
-/* String manipulation functions. */
+/** \file strings.nix
+ * String manipulation functions.
+ */
 { lib }:
 let
 
 inherit (builtins) length;
-
+/** @private */
```

Comment:
notice how I had to mark this as private. since doxygen is in c++ mode it thinks these are variables.